src/sqlancer/Main.java

```diff
@@ -312,30 +345,30 @@ public void run() throws Exception {
                 if (options.logEachSelect()) {
                     logger.writeCurrent(state.getState());
                 }
-                provider.generateAndTestDatabase(state);
                 try {
-                    logger.getCurrentFileWriter().close();
-                    logger.currentFileWriter = null;
-                } catch (IOException e) {
-                    throw new AssertionError(e);
+                    provider.generateAndTestDatabase(state);
+
+                } catch (FoundBugException e) {
+                    try {
+                        logger.getCurrentFileWriter().close();
```

Comment:
This seems quite hacky (perhaps part of my initial prototype implementation), not sure if there is a better way to implement this functionality.

Reply:
TBH, I don't understand enough of how Main works. This is taken from your original prototype, and I'm open to suggestions for how to do it better!